utils/analytics.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  getAnalytics,
+  getAppInstanceId,
+  logEvent,
+  setAnalyticsCollectionEnabled,
+  logScreenView,
+} from '@react-native-firebase/analytics';
+import { getApp } from '@react-native-firebase/app';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+let analytics: ReturnType<typeof getAnalytics> | null = null;
+try {
+  const app = getApp();
+  analytics = getAnalytics(app);
+} catch (error) {
+  // Analytics initialization failed - app will continue without analytics
+}
+const sanitize = (value?: string): string =>
+  typeof value === 'string' && value.trim().length > 0 ? value.trim() : 'undefined';
+
+const isAnalyticsReady = (): boolean => !!analytics;
+
+const checkConsent = async (): Promise<boolean> => {
+  try {
+    const consent = await AsyncStorage.getItem('consent');
+    return consent === null ? true : consent === 'true';
```

Comment:
This is very confusing, why didn't we just return the value `consent` ?

Reply:
And is there an issue in pushing the consent value on first run of the app? How about if set the consent value to true when user opens the app for first time.